lib/Parse/ParseIfConfig.cpp

```diff
@@ -25,6 +26,496 @@
 
 using namespace swift;
 
+namespace {
+
+/// Get PlatformConditionKind from platform condition name.
+static
+Optional<PlatformConditionKind> getPlatformConditionKind(StringRef Name) {
+  return llvm::StringSwitch<llvm::Optional<PlatformConditionKind>>(Name)
+    .Case("os", PlatformConditionKind::OS)
+    .Case("arch", PlatformConditionKind::Arch)
+    .Case("_endian", PlatformConditionKind::Endianness)
+    .Case("_runtime", PlatformConditionKind::Runtime)
+    .Default(None);
+}
+
+/// Extract source text of the expression.
+static StringRef extractExprSource(SourceManager &SM, Expr *E) {
+  CharSourceRange Range =
+    Lexer::getCharSourceRangeFromSourceRange(SM, E->getSourceRange());
+  return SM.extractText(Range);
+}
+
+/// Get the identifier string of the UnresolvedDeclRefExpr.
+/// Returns \c true if failed.
+static bool getNameStr(Expr *E, StringRef &Name, DeclRefKind Kind) {
```

Comment:
My only nitpick: I think this could be made to return an `Optional<StringRef>` - and probably could have a more descriptive name.  Right now it matches a member of `ValueDecl` but does something more specific than `ValueDecl`'s implementation.